Modules/_io/bufferedio.c

```diff
@@ -1817,10 +1839,34 @@ _io_BufferedWriter___init___impl(buffered *self, PyObject *raw,
     self->fast_closed_checks = (Py_TYPE(self) == &PyBufferedWriter_Type &&
                                 Py_TYPE(raw) == &PyFileIO_Type);
 
+    if (self->buffered_writers_list.next == NULL) {
+        self->buffered_writers_list.prev = &doubly_linked_end;
+        self->buffered_writers_list.next = doubly_linked_end.next;
+        doubly_linked_end.next->prev = &self->buffered_writers_list;
+        doubly_linked_end.next = &self->buffered_writers_list;
+    }
+
     self->ok = 1;
     return 0;
 }
 
+/*
+* Ensure all buffered writers are flushed before proceeding with
+* normal shutdown.  Otherwise, if the underlying file objects get
+* finalized before the buffered writer wrapping it then any buffered
+* data will be lost.
+*/
+void _PyIO_atexit_flush(void)
+{
+    while (doubly_linked_end.next != &doubly_linked_end) {
+        buffered *buf = (buffered *)(((char *)doubly_linked_end.next) -
+                                     offsetof(buffered, buffered_writers_list));
+        remove_from_linked_list(buf);
+        buffered_flush(buf, NULL);
+        PyErr_Clear();
```

Comment:
Rather than silence errors, they can be displayed using `PyErr_WriteUnraisable`.